lib/repl.js

```diff
@@ -844,10 +845,61 @@ REPLServer.prototype.createContext = function() {
   });
 
   addBuiltinLibsToObject(context);
+  previewResults.call(this);
 
   return context;
 };
 
+function previewResults() {
+  this.on('buffer', function(line) {
+
+    Interface.prototype.clearPreview.call(this);
+
+    function isValidExpression(code) {
+      try {
+        const type = parse(code.toString()).body[0].expression.type;
+        return type.includes('Expression') || type === 'Identifier';
+      } catch (e) {
+        return false;
+      }
+    }
+
+    try {
+      if (isValidExpression(line)) {
```

Comment:
there's no need to check if its a valid expression, or even an expression at all.

Reply:
Any reasons for that? You mean we can just wrap them in a try catch block and that should do good for us?

Reply:
I just thought sending code for every key press for eval might be unnecessary.